pkg/apis/cr/v1alpha1/repositoryserver_types.go

```diff
@@ -117,48 +117,44 @@ type UserAccess struct {
 
 // RepositoryServerStatus is the status for the RepositoryServer. This should only be updated by the controller
 type RepositoryServerStatus struct {
-	Conditions []Condition              `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	Conditions []metav1.Condition       `json:"conditions,omitempty"`
 	ServerInfo ServerInfo               `json:"serverInfo,omitempty"`
 	Progress   RepositoryServerProgress `json:"progress,omitempty"`
 }
 
-// Condition contains details of the current state of the RepositoryServer resource
-type Condition struct {
-	LastTransitionTime metav1.Time                   `json:"lastTransitionTime,omitempty"`
-	LastUpdateTime     metav1.Condition              `json:"lastUpdateTime,omitempty"`
-	Status             metav1.ConditionStatus        `json:"status"`
-	Type               RepositoryServerConditionType `json:"type"`
-}
+const (
 
-// RepositoryServerConditionType defines all the various condition types of the RepositoryServer resource
-type RepositoryServerConditionType string
+	// ServerSetup indicates whether the repository pod and service have been
+	ServerSetup string = "ServerSetup"
+
+	// RepositoryConnected indicates whether the existing repository is connected and ready to use
+	RepositoryConnected string = "RepositoryConnected"
 
-const (
 	// RepositoryReady indicates whether the existing repository is connected and ready to use
-	RepositoryReady RepositoryServerConditionType = "RepositoryReady"
+	RepositoryReady string = "RepositoryReady"
 
 	// ServerInitialized means that the proxy server, that serves the repository, has been started
-	ServerInitialized RepositoryServerConditionType = "ServerInitialized"
+	ServerInitialized string = "ServerInitialized"
 
 	// ClientsInitialized indicates that the client users have been added or updated to the repository server
-	ClientsInitialized RepositoryServerConditionType = "ClientsInitialized"
+	ClientsInitialized string = "ClientsInitialized"
```

Comment:
we decided to change this to `ClientUserInitialized` right?

Reply:
Right. I missed it somehow. Have updated it now.